src/OrchardCore.Modules/OrchardCore.Users/Extensions/UserTimeZoneServiceExtensions.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Modules;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.TimeZone.Services;
+
+namespace OrchardCore.Users;
+
+/// <summary>
+/// Provides an extension for <see cref="IUserTimeZoneService"/>.
+/// </summary>
+public static class UserTimeZoneServiceExtensions
+{
+    private static HttpContext HttpContext => new HttpContextAccessor().HttpContext;
```

Comment:
@hishamco never do this. You should never hold the HttpContext object like this in a static property.

Reply:
Right, I was thinking to find a way to get the right `HttpContext`

Reply:
you should not worry about that. Either pass the HttpContext as an argument or remove the extension.